fsspec/implementations/local.py

```diff
@@ -59,11 +59,19 @@ def rmdir(self, path):
 
     def ls(self, path, detail=False, **kwargs):
         path = self._strip_protocol(path)
-        if detail:
+        if not os.path.exists(path):
+            raise FileNotFoundError(path)
```

Comment:
I'm guessing I should also add `errno.ENOENT` here to get a proper `No such file or directory` message.

Reply:
I don't think that's necessary.